### PR TITLE
[INFRA] Add structured logging linter script to github workflow

### PR DESCRIPTION
**Pull request by zedtang**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [x] Other (Infra)

## Description

Add a python linter script to enforce new code to adhere to structured logging. There is a similar effort in Spark: https://github.com/apache/spark/pull/47239.
## How was this patch tested?

github workflow passed.
## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

File: .github/workflows/spark_test.yaml [hunk 1]
```diff
@@ -80,6 +80,11 @@ jobs:
           pipenv run pip install pyarrow==8.0.0
           pipenv run pip install numpy==1.20.3
         if: steps.git-diff.outputs.diff
+      - name: Scala structured logging check
```
Comment (scottsand-db) [inline .github/workflows/spark_test.yaml:83]: "check" is ambiguous. "style check"?

File: dev/structured_logging_style.py
Comment (scottsand-db) [inline dev/structured_logging_style.py:1]: Rename to `spark_structured_logging_style.py`

File: .github/workflows/spark_test.yaml (hunk 1 above)
Comment (zedtang) [reply .github/workflows/spark_test.yaml]: updated

File: dev/structured_logging_style.py
Comment (zedtang) [reply dev/structured_logging_style.py:1]: done

**Review by gengliangwang — APPROVED** [review]

**Review by scottsand-db — APPROVED** [review]

**Pull request merged**